libraries/lib-xml/XMLAttributeValueView.h

```diff
@@ -0,0 +1,174 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  XMLAttributeValueView.h
+
+  Dmitry Vedenko
+
+**********************************************************************/
+
+#pragma once
+
+#include <cstdint>
+#include <string>
+#include <string_view>
+#include <wx/string.h>
+
+/*! \brief A view into an attribute value. The class does not take the ownership of the data.
+ *
+ * Audacity internally uses regular XML and an internal binary XML-like format.
+ * Binary format extends the possible XML types to support integers and floating point values,
+ * not just the string values.
+ *
+ * This class represents the attribute value for both test and binary XML formats.
+ *
+ * Class has a number of the Get overloads, that allow to read the value with the required data type.
+ *
+ * Integer-based types can be read from the view if it was initialized by an integer value or with a
+ * string_view that is a textual representation of an integer value. To avoid runtime data loss it is not
+ * possible to read an integer from the floating point values. Boundary checks are always performed.
+ * If the type is not large enough to store the value, Get fails.
+ *
+ * Floating point values can be read from the view, if it is initialized using floating point, integer or
+ * a compatible string value. Conversion from Float to Double is forbidden.
+ *
+ * String values can be read only if the view was created using a string_view.
+ * However there are ToString() and ToWString() methods that allow conversion from any type to string.
+ * Null type is represented by an empty string.
+ */
+class XML_API XMLAttributeValueView final
+{
+public:
+   //! Type of the value represented by the XMLAttributeValueView.
+   enum class Type
+   {
+      Null, //!< The value is not initialized.
+      SignedInteger, //!< The value is a signed integer.
+      UnsignedInteger, //!< The value is an unsigned integer.
+      Float, //!< The value is a single precision floating point value.
+      Double, //!< The value is a double precision floating point value.
+      StringView //!< The value is a string_view.
+   };
+
+   //! Construct an unitialized view of type Null
+   XMLAttributeValueView() = default;
+
+   XMLAttributeValueView(const XMLAttributeValueView&) = default;
+   XMLAttributeValueView(XMLAttributeValueView&&) = default;
+
+   XMLAttributeValueView& operator = (const XMLAttributeValueView&) = default;
+   XMLAttributeValueView& operator = (XMLAttributeValueView&&) = default;
+
+   //! Construct a view of type SignedInteger from the value.
+   explicit XMLAttributeValueView(bool value) noexcept;
```

Comment:
std::is_signed_v<bool> is false:  https://en.cppreference.com/w/cpp/types/is_signed

So maybe you should do like the standard library?

Reply:
Sure, why not!